fireworks/core/launchpad.py

```diff
@@ -1269,8 +1291,28 @@ def complete_launch(self, launch_id, action=None, state='COMPLETED'):
         if action:
             m_launch.action = action
 
-        self.launches.find_one_and_replace({'launch_id': m_launch.launch_id}, m_launch.to_db_dict(),
-                                           upsert=True)
+        try:
+            self.launches.find_one_and_replace({'launch_id': m_launch.launch_id},
+                                               m_launch.to_db_dict(), upsert=True)
+        except DocumentTooLarge:
+            launch_db_dict = m_launch.to_db_dict()
+            action_dict = launch_db_dict.get("action", None)
+            if self.gridfs_fallback is None or not action_dict:
```

Comment:
these seem like 2 different errors

If there's no action_dict, reraising the error seems appropriate

If there is an action_dict but the user has not set gridfs_fallback, perhaps you should add to the error that the user should set a gridfs fallback collection

Reply:
(or at least raise a warning that the gridfs fallback was not set, but the document is too large)